components/suggestion/suggestion_box/suggestion_box.jsx

```diff
@@ -638,9 +638,10 @@ export default class SuggestionBox extends React.PureComponent {
         const terms = suggestions.terms;
         const items = suggestions.items;
         let selection = this.state.selection;
-        if (terms.length > 0) {
+        const selectionIndex = terms.indexOf(selection);
+        if (terms.length > 0 && selectionIndex === -1) {
             selection = terms[0];
-        } else if (this.state.selection) {
+        } else if (this.state.selection && selectionIndex === -1) {
```

Comment:
This can probably be simplified to make it simpler to reason about. Anyway non-blocking.
```js
if (terms.indexOf(selection) === -1) {
    if (terms.length > 0) {
        selection = terms[0];
    } else {
        selection = '';
    }
```

Reply:
Yeah, I think I need to track the selection index one test case is failing due to this change